src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/IBotActionData.cs

```diff
@@ -1,14 +1,31 @@
-﻿using System.Text;
+﻿using System.Collections.Generic;
+using System.Text;
+using RegressionGames.StateRecorder.Models;
 
 namespace RegressionGames.StateRecorder.BotSegments.Models
 {
     public interface IBotActionData
     {
+
+        /**
+         * Called before the first call to ProcessAction to allow data setup by the action
+         */
+        public void StartAction(int segmentNumber, Dictionary<int, TransformStatus> currentUITransforms, Dictionary<int, TransformStatus> currentGameObjectTransforms);
+
+        /**
+         * Called at least once per frame
```

Comment:
What decides if this is called more than once per frame?

Reply:
updated comment to once per frame

in one of the early implementations this could be called 2x for the same segment, but since Time.unscaledTime only updates once per frame, that was pointless